scripts/_bouncer.sh

```diff
@@ -0,0 +1,208 @@
+config_not_set() {
+    require 'CONFIG'
+    local varname before after
+
+    varname=$1
+    if [ "$varname" = "" ]; then
+        msg err "missing required variable name"
+        exit 1
+    fi
+
+    before=$("$BOUNCER" -c "$CONFIG" -T)
+    # shellcheck disable=SC2016
+    after=$(echo "$before" | envsubst "\$$varname")
+
+    if [ "$before" = "$after" ]; then
+        return 1
+    fi
+    return 0
+}
+
+need_api_key() {
+    if config_not_set 'API_KEY'; then
+        return 0
+    fi
+    return 1
+}
+
+# Interpolate a variable in the config file with a value.
+set_config_var_value() {
+    require 'CONFIG'
+    local varname value before
+
+    varname=$1
+    if [ "$varname" = "" ]; then
+        msg err "missing required variable name"
+        exit 1
+    fi
+
+    value=$2
+    if [ "$value" = "" ]; then
+        msg err "missing required variable value"
+        exit 1
+    fi
+
+    before=$(cat "$CONFIG")
+    echo "$before" | \
+        env "$varname=$value" envsubst "\$$varname" | \
+        install -m 0600 /dev/stdin "$CONFIG"
+}
+
+set_api_key() {
+    require 'CONFIG' 'BOUNCER_PREFIX'
+    local api_key ret bouncer_id before
+    # if we can't set the key, the user will take care of it
+    ret=0
+
+    if command -v cscli >/dev/null; then
+        echo "cscli/crowdsec is present, generating API key" >&2
+        bouncer_id="$BOUNCER_PREFIX-$(date +%s)"
+        api_key=$(cscli -oraw bouncers add "$bouncer_id" || true)
+        if [ "$api_key" = "" ]; then
+            echo "failed to create API key" >&2
+            api_key="<API_KEY>"
+            ret=1
+        else
+            echo "API Key successfully created" >&2
+            echo "$bouncer_id" > "$CONFIG.id"
+        fi
+    else
+        echo "cscli/crowdsec is not present, please set the API key manually" >&2
+        api_key="<API_KEY>"
+        ret=1
+    fi
+
+    if [ "$api_key" != "" ]; then
+        set_config_var_value 'API_KEY' "$api_key"
+    fi
+
+    return "$ret"
+}
+
+set_local_port() {
+    require 'CONFIG'
+    local port
+    command -v cscli >/dev/null || return 0
+    # the following will fail with a non-LAPI local crowdsec, leaving empty port
+    port=$(cscli config show --key "Config.API.Server.ListenURI" 2>/dev/null | cut -d ":" -f2 || true)
+    if [ "$port" != "" ]; then
+        sed -i "s/localhost:8080/127.0.0.1:$port/g" "$CONFIG"
+        sed -i "s/127.0.0.1:8080/127.0.0.1:$port/g" "$CONFIG"
+    fi
+}
+
+set_local_lapi_url() {
+    require 'CONFIG'
+    local port before varname
+    # $varname is the name of the variable to interpolate
+    # in the config file with the URL of the LAPI server,
+    # assuming it is running on the same host as the
+    # bouncer.
+    varname=$1
+    if [ "$varname" = "" ]; then
+        msg err "missing required variable name"
+        exit 1
+    fi
+    command -v cscli >/dev/null || return 0
+
+    port=$(cscli config show --key "Config.API.Server.ListenURI" 2>/dev/null | cut -d ":" -f2 || true)
```

Comment:
please force raw output of cscli, see https://github.com/crowdsecurity/cs-custom-bouncer/pull/91/commits/e30ff3fbf93d037ff7a649feae1d7a2c2c96eaf1

Reply:
fixed via https://github.com/crowdsecurity/cs-cloudflare-worker-bouncer/pull/1/commits/b5850a6415bbdc35dcd039b51cbc8ab55d7bf409